README.md

```diff
@@ -1,67 +1,153 @@
+
+## Features
+
+- Deposit via direct transfer
+- Deposit via an allowance
+- Withdrawal mechanism
+- User balance tracking
 
-### Links
+## Links
 
-The package is published on [MOPS](https://mops.one/token-handler) and [GitHub](https://github.com/research-ag/token-handler).
-Please refer to the README on GitHub where it renders properly with formulas and tables.
+The package is published on [MOPS](https://mops.one/token-handler) and [GitHub](https://github.com/research-ag/token-handler).
 
-The API documentation can be found [here](https://mops.one/token-handler/docs/lib) on Mops.
+The API documentation can be found [here](https://mops.one/token-handler/docs/lib) on Mops.
 
 For updates, help, questions, feedback and other requests related to this package join us on:
 
 - [OpenChat group](https://oc.app/2zyqk-iqaaa-aaaar-anmra-cai)
 - [Twitter](https://twitter.com/mr_research_ag)
 - [Dfinity forum](https://forum.dfinity.org/)
 
-### Motivation
+## Build & test
 
-The package was developed to streamline and secure the entire lifecycle of user funds on an ICRC1 ledger, including deposits, credits, and withdrawals.
+We need up-to-date version of `mops` installed.
 
-## Usage
+Then run:
+
+```
+git clone git@github.com:research-ag/token-handler.git
+mops install
+mops test
+```
 
-### Install with mops
+## Installation
 
-You need `mops` installed. In your project directory run:
+You need `mops` installed. In your project directory run:
 
 ```
 mops add token-handler
 ```
 
 In the Motoko source file import the package as:
 
-```motoko
+```
 import TokenHandler "mo:token-handler";
 ```
 
-### Example
+## Example
 
-To see token-handler package in action, check out the [example code](https://github.com/research-ag/token-handler/blob/main/example).
+To see token-handler package in action, check out the [example code](https://github.com/research-ag/token-handler/blob/main/example). The example is a simple implementation of the [ICRC-84](https://github.com/research-ag/icrc-84) standard.
 
-### Build & test
+## Documentation
 
-We need up-to-date versions of `node`, `moc` and `mops` installed.
-Suppose `<path-to-moc>` is the path of the `moc` binary of the appropriate version.
+### Requirements
 
-Then run:
+The only requirement on the underlying token ledger is the ICRC-1 standard. Deposit method via direct transfer is available without ICRC-2 extension. For using the deposit method via an allowance ICRC-2 extension is required.
 
-```
-git clone git@github.com:research-ag/token-handler.git
-mops install
-DFX_MOC_PATH=<path-to-moc> mops test
-```
+It is not required that the service can inspect individual deposit transactions by transaction id, memo or other means. Hence, it is not required that the underlying token ledger provides an indexer, transaction history or archive. In particular, the ICRC-3 extension is not required.
```

Comment:
We should probably say somewhere that a service can support multiple tokens but has to operate a separate instance of TokenHandler for each supported token.